Configurations/15-android.conf

```diff
@@ -79,7 +91,7 @@
                 (my $tridefault = $triarch) =~ s/^arm-/$arm-/;
                 (my $tritools   = $triarch) =~ s/(?:x|i6)86(_64)?-.*/x86$1/;
                 $cflags .= " -target $tridefault "
-                        .  "-gcc-toolchain \$(ANDROID_NDK)/toolchains"
+                        .  "-gcc-toolchain $ndk/toolchains"
```

Comment:
Rationale behind keeping $(ANDROID_NDK) (as opposite to "freezing" it at config time for example as suggested above) was to *maximize* probability of failure in case 'make' is executed in different window. It might sound counter-intuitive, in sense that normally you would tend to minimize probability of failures, but intention was to minimize something else, namely probability of user ending up in unsupported situation. For give an example. Let's say you aim for clang, adjust PATH and ANDROID_NDK in one window, run 'Configure', and then 'make' in another. If NDK location was frozen in Makefile, 'make' is likely to actually work in that other window. Because non-NDK clang that is now likely to be found on non-adjusted PATH can actually generate non-x86 code. Even resulting binary code might work, but thing is that we don't want to end up in situation when something doesn't as result of this. For this reason it's deemed desirable to encourage user to stick with clang bundled with NDK.

[Just in case, one can wonder who does that, adjusts PATH and sets ANDROID_NDK in just one window, and not in say $HOME/.profile. Well, those who target multiple platforms, switch between contexts, even between different compilers and their versions, different NDKs... BTW, yes, arrangement is not 100% fool-proof, because you can still set ANDROID_NDK alone in the other window and end up with non-NDK clang, but at least you are not encouraged to do so.]

Reply:
Yes, I agree that if you do ./Configure in one window then make should work with that config in another window. I think this PR does exactly this.

My goal is that `I would like ./Configure target=aarch64-linux-android && make` just works without any special handling of environment variables.
Support for cross-compilation in Configure should be improved and e.g. switching from gcc to clang is still to complicated. (e.g. removing -mandroid flag)

Reply:
> I agree that if you do ./Configure in one window then make should work

??? What do you mean "agree"? With whom? What I'm saying is that intention was to *prevent* it from working, i.e. opposite to what you agree with.

Reply:
Then it seems I misunderstood your previous comment and then we disagree of course.
I think that Configure should freeze the configuration. It does not currently - at least not as I need it.
Sometimes I need to build different libraries for different architectures e.g. libsodium and libcrypto (openssl) for arm and aarch64. This is really too easy to screw up currently. So what I and others do in this case is write scripts to setup the environment for each build that then call `configure --prefix=/usr/local/$triplet/ && make`. I would like to remove the need for this script writing so that in the openssl case 
`./Configure --target=triplet --prefix=/usr/local/triplet && make && make install && make clean` 
`./Configure --target=other-triplet --prefix=/usr/local/other-triplet && make && make install` 
just works (in the same window).

Reply:
> just works (in the same window)

If you play your cards right, it does. If you have binutils-multiarch and have NDK's clang on $PATH, you can `./Configure android-arm && make` followed by `make distclean && ./Configure android-arm64 && make`. In the same window. Well, modulo #7443... [I failed to test it implicit API level, because I customarily test with `-D__ANDROID_API__=21 -static`, so that I can run `make test`]

(Just in case, I [for one] wouldn't mind freezing NDK location if one could also freeze PATH in Makefile.)

Reply:
Couldn't this PATH-freezing be done by instead of setting like
`CROSS_COMPILE=arm-linux-androideabi-`
setting it to
`CROSS_COMPILE=${config_time_path}/arm-linux-androideabi-`